CONTRIBUTING.adoc

```diff
@@ -0,0 +1,74 @@
+= Contributing to the mobile cli
+
+This document explains how to set up a development environment and get involved with Mobile CLI project.
+
+Before anything else, https://help.github.com/articles/fork-a-repo/[fork] the Mobile CLI project.
+
+== Tools We use
+
+* Git
+* Go
+* Make
+
+== Set Up OpenShift
+
+The Mobile CLI targets Kubernetes and is intended to be developed against a running Kubernetes cluster,
+we use OpenShift as our Kuberentes distribution. The Mobile CLI is intended to help you work with Mobile Services running ontop of OpenShift.
+To provision these services, we levarage the https://github.com/kubernetes-incubator/service-catalog[Service Catalog] and the https://github.com/openshift/ansible-service-broker[Ansible Service Broker].
+To help get this infrastructure set up there is a https://github.com/aerogear/mobile-core#installing-from-a-development-release[ansible based installer] provided.
+
+== Clone the repositiory
+
+As we are using Go, the path you clone this repo into is important.
+
+* create the directory `mkdir -p $GOPATH/src/github.com/aerogear`
+* clone the repo `cd $GOPATH/src/github.com/aerogear &amp;&amp; git clone git@github.com:aerogear/mobile-cli.git`
+* add your own fork as the upstream target `git add remote upstream &lt;your fork&gt;`
+
+== Building the Mobile CLI
+
+To build the cli locally you can run `make build` this command will run a set of checks and the unit tests before compiling the binary and outputting it into the current directory,
+if you only want to build the binary itself you can simply run `make build_binary`.
+Once built, you can access this binary and use it from the command line `./mobile`
+Remember however that the CLI is intended as a Kubernetes plugin so expects to kind kube configuration in `~/.kube/config`. If you have setup OpenShift this should
+already be in place.
+
+== Submitting changes to the Mobile CLI
+
+=== Before making a pull request
+
+There are a few things you should keep in mind before creating a PR.
+
+* New code should have corresponding unit tests. An example of how we approach unit testing can be found in https://github.com/aerogear/mobile-cli/blob/master/pkg/cmd/clients_test.go[clients_test.go].
+
+* Ensure for new commands to read the https://github.com/aerogear/mobile-cli/doc/adding_new_cmd.md[adding a new command] doc before hand.
```

Comment:
Might be clearer phrased as:
`When adding a new command, be sure to read the ....`